flumine/order/process.py

```diff
@@ -79,11 +79,15 @@ def process_current_orders(
 def process_current_order(order: BaseOrder, current_order, log_control) -> None:
     # update
     order.update_current_order(current_order)
-    # pickup async orders
-    if order.async_ and order.bet_id is None and current_order.bet_id:
-        order.responses.placed()
-        order.bet_id = current_order.bet_id
-        log_control(OrderEvent(order))
+    if order.bet_id is None and current_order.bet_id:
+        # pickup async orders
+        if order.async_:
+            order.responses.placed()
+            order.bet_id = current_order.bet_id
+            log_control(OrderEvent(order))
+        # pickup FOK orders
```

Comment:
That is a good point. It is provided by Betfair: https://github.com/betcode-org/flumine/issues/681#issuecomment-1679635017

Reply:
This seems to happen when the order comes back in the stream before we get the instruction report and set the bet_id. Just tested and think the fix in betfairexecution.py should be enough.

Reply:
That would explain the stochastic nature of this bug. However, I am quite surprised this has not been picked up on earlier because it's not just FOK bets that are affected by the place response arriving later than the order update.